server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java

```diff
@@ -114,4 +123,14 @@ protected void appendThreadPoolExecutorDetails(StringBuilder sb) {
             .append(", ");
     }
 
+    /**
+     * Returns the set of currently running tasks and their start timestamp.
+     * <p>
+     * Note that it is possible for a task that has just finished execution to be temporarily both in the returned map, and its total
+     * execution time to be included in the return value of {@code getTotalTaskExecutionTime()}. However, it is guaranteed that the
+     * task is reflected in at least one of those two values.
+     */
+    public Map<Runnable, Long> getOngoingTasks() {
```

Comment:
Note that for the estimation we do not need the tasks counted/considered in `ongoingTasks` and `totalExecutionTime` to be exclusive. This also helps to avoid unnecessary synchronization.